File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
      int left = 0;
      int right = 1;
      int maxS = 1;

      while (right < s.size()){
        if(s[left] == s[right]){
          left++;
        }else{
          right++;
        }
        maxS = max(maxS,right - left);
      }
      return maxS;

    }
};

File: Data Structures & Algorithms/house-robber-ii/submission-1.py
class Solution:
    def rob(self, nums: list[int]) -> int:
        n = len(nums)
        nums1 = nums[:n+1]
        nums2 = nums[1:]
        def dfs(i, arr):
            if i >= len(arr):
                return 0
            
            return max(arr[i] + dfs(i+2,arr),dfs(i+1,arr))
        left = dfs(0,nums1)
        right = dfs(0,nums2)
        return min(left,right)
